IN-CORE/pyincore
# 404 support local hazard

This is ready to be reviewed. This is a huge PR with the below features:

1. Create parent class **Hazard** and child classes **Earthquake**, **Tornado**, **Tsunami**, **Flood** and **Hurricane**
2. Represent hazard dataset in combine with existing Dataset class. Created **HazardDataset** and child classes accordingly
3. Backward compatibility of instantiate hazard from **remote service**
4. Create hazard **from local file**
5. **Attach local dataset** (raster and shapefile) to the locally created hazard
6. Read raster values; calculate tornado wind speed from shapefile. The response is made identical to the response from services
7. Lastly, update the **Bridge Damage** as an example to demonstrate how to integrate it with analysis

To Test:
1. Run pytest **tests/pyincore/models/test_units.py**. --> it covers almost all scenarios
2. Run Bridge Damage test e.g. **test_bridgedamage.py**, or **test_mmsa_bridgedamage.py** and compare the results with some previous. See attached zip file
[legacy.zip](https://github.com/IN-CORE/pyincore/files/12580845/legacy.zip)

3. Run local hurricane test **test_bridgedamage_w_local_hazard.py** and compare with galveston bridge damage result

To Dos:
1. Double check model based hazard; I might miss a few fields (but none of the fields are used anyway except id)
2. For EQ, need to add logic to find nearest period --> create new task
3. For EQ, need to add logic to convert pga to pgv, sd, sv, etc --> create new task 
5. Include more unit conversion especially for wind speed --> create new task
